cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala

```diff
@@ -37,114 +33,38 @@ class DrsLocalizerMain(drsUrl: String,
-             |  if grep -q '$RequesterPaysErrorMsg' gsutil_output.txt; then
-             |    echo "Received 'Bucket is requester pays' error. Attempting again using Requester Pays billing project"
-             |    ${gcsCopyCommand(s"-u $userProject")} > gsutil_output.txt 2>&1
-             |    RC_GSUTIL=$$?
-             |  fi
-             |fi""".stripMargin
-        case None => ""
-      }
-    }
-
-    // bash to download the GCS file using gsutil
-    s"""set -euo pipefail
-       |set +e
-       |
-       |${setServiceAccount()}
-       |
-       |# Run gsutil copy without using project flag
-       |${gcsCopyCommand()} > gsutil_output.txt 2>&1
-       |RC_GSUTIL=$$?
-       |
-       |${recoverWithRequesterPays()}
-       |
-       |if [ "$$RC_GSUTIL" != "0" ]; then
-       |  echo "Failed to download the file. Error: $$(cat gsutil_output.txt)" >&2
-       |  exit "$$RC_GSUTIL"
-       |else
-       |  echo "Download complete!"
-       |  exit 0
-       |fi""".stripMargin
-  }
-
-
-  def downloadFileFromGcs(gcsUrl: String, serviceAccountJsonOption: Option[String]) : IO[ExitCode] = {
-
-    logger.info(s"Requester Pays project ID is $requesterPaysProjectIdOption")
-    logger.info(s"Attempting to download $gcsUrl to $downloadLoc")
-
-    val copyProcess = serviceAccountJsonOption match {
-      case Some(sa) =>
-        // if Martha returned a SA, use that SA for gsutil instead of default credentials
-        val tempCredentialDir: Path = Files.createTempDirectory("gcloudTemp_").toAbsolutePath
-        val saJsonPath: Path = tempCredentialDir.resolve("sa.json")
-        Files.write(saJsonPath, sa.getBytes(StandardCharsets.UTF_8))
-        val extraEnv = Map("CLOUDSDK_CONFIG" -> tempCredentialDir.toString)
-        val copyCommand = Seq("bash", "-c", downloadScript(gcsUrl, Option(saJsonPath)))
-        Process(copyCommand, None, extraEnv.toSeq: _*)
-      case None =>
-        // No SA returned from Martha. gsutil will use the application default credentials.
-        val copyCommand = Seq("bash", "-c", downloadScript(gcsUrl, None))
-        Process(copyCommand)
-    }
-
-    // run the multiple bash script to download file and log stream sent to stdout and stderr using ProcessLogger
-    val returnCode = copyProcess ! ProcessLogger(logger.underlying.info, logger.underlying.error)
-
-    IO(ExitCode(returnCode))
-  }
-
-
-  def resolveAndDownload(): IO[ExitCode] = {
-    val fields = NonEmptyList.of(MarthaField.GsUri, MarthaField.GoogleServiceAccount)
+  def resolve(): IO[Downloader] = {
+    val fields = NonEmptyList.of(MarthaField.GsUri, MarthaField.GoogleServiceAccount, MarthaField.AccessUrl)
     for {
-      localizerGcsDrsPathResolver <- getGcsDrsPathResolver
-      marthaResponse <- localizerGcsDrsPathResolver.resolveDrsThroughMartha(drsUrl, fields)
-      // Currently Martha only supports resolving DRS paths to GCS paths
-      gcsUrl <- IO.fromEither(marthaResponse.gsUri.toRight(new RuntimeException(ExtractGcsUrlErrorMsg)))
```

Comment:
Did we lose this check along the way? The code is gone, but maybe it's rephrased somewhere and I'm not recognizing it.

Reply:
This condition now maps to [this line](https://github.com/broadinstitute/cromwell/blob/bt_186_bdc_public_drs_localizer_thin_slice/cromwell-drs-localizer/src/main/scala/drs/localizer/DrsLocalizerMain.scala#L65) where we have neither an access URL nor a GS URI.